teuthology/dispatcher/supervisor.py

```diff
@@ -151,6 +153,46 @@ def run_job(job_config, teuth_bin_path, archive_dir, verbose):
     if 'targets' in job_config:
         unlock_targets(job_config)
 
+def failure_is_reimage(failure_reason):
+    if not failure_reason:
+        return False
+    reimage_failure = "Error reimaging machines:"
+    if reimage_failure in failure_reason:
+        return True
+    else:
+        return False
+
+def check_for_reimage_failures_and_mark_down(targets, count=10):
+    # Grab paddles history of jobs in the machine
+    # and count the number of reimaging errors
+    # if it fails N times then mark the machine down
+    base_url = teuth_config.results_server
+    for k, _ in targets.items():
+        machine = k.split('@')[-1]
+        url = urljoin(
+                base_url,
+                '/nodes/{0}/jobs/?count={1}'.format(
+                machine, count)
+        )
+        resp = requests.get(url)
+        jobs = resp.json()
+        if len(jobs) < count:
+            continue
+        reimage_failures = list(filter(
+            lambda j: failure_is_reimage(j['failure_reason']),
+            jobs
+        ))
+        if len(reimage_failures) < count:
+            continue
+        # Mark machine down
+        machine_name = machine.split('.')[0]
```

Comment:
This shouldn't be necessary

Reply:
Think it is necessary since we are comparing the count of the failures to determine if we should mark the machines down or not as well as we do the split because we need only the machine_name to execute the markdown command.

Reply:
Changed to using `decanonicalize_hostname` in `teuthology.misc` for getting the `machine_name`